src/expr/src/relation/mod.rs

```diff
@@ -3077,13 +3080,6 @@ impl RowSetFinishing {
                         .extend(self.project.iter().map(|i| &datums[*i]));
                     row_buf.clone()
                 };
-                total_bytes += new_row.byte_len();
-                if total_bytes > max_result_size {
-                    return Err(format!(
-                        "result exceeds max size of {}",
-                        ByteSize::b(u64::cast_from(max_result_size))
-                    ));
-                }
```

Comment:
If this is getting removed, what actually causes us to return early in the case where the result exceeds the max size?

Reply:
Nevermind, I see it's in the earlier loop.

Reply:
I think the theory is that the earlier return near the other modified code would do the return, and this code is "dead" in that any time this would return, that earlier code would also return.

It would be a lot easier to understand without two loops doing the same logic, though.

Reply:
The earlier block is documented as returning early if we would exceed the limit, and assuming it holds to that documentation this code should not have to return early in that case.

Reply:
I read this more, and @mjibson I think there's a gotcha here that this accounting respects `self.project` and the accounting up above does not. It's .. probably just a bug that the one above does not respect `self.project`, but it does mean that these could reach different conclusions.

Reply:
Looks like we want to avoid a panic in `Vec::with_capacity`: https://github.com/MaterializeInc/materialize/pull/17968